src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/ConnectionHandlerContext.cs

```diff
@@ -31,5 +31,40 @@ internal class ConnectionHandlerContext : HandlerRequest
         /// Class used by orchestrator while chaining handlers.
         /// </summary>
         public Exception Error { get; set; }
+
+        /// <summary>
+        /// Method needed to clone ConnectionHandlerContext as part of history along the chain.
+        /// </summary>
+        public object Clone() 
+        {
+            ConnectionHandlerContext clonedContext = new ConnectionHandlerContext();
+
+            if (this.ConnectionStream != null)
+            {
+                MemoryStream clonedStream = new MemoryStream();
+                this.ConnectionStream.CopyTo(clonedStream);
+                clonedStream.Position = 0;
+                clonedContext.ConnectionStream = clonedStream;
+            }
+
+            if (this.ConnectionString != null)
+            {
+                clonedContext.ConnectionString = new SqlConnectionString(this.ConnectionString.ToString());
+            }
+
+            if (this.DataSource != null)
+            {
+                //TODO: Move DataSource class into SqlClientX and implement clone
+                clonedContext.DataSource = (DataSource)this.DataSource;
+            }
+
+            if (this.Error != null)
+            {
+                clonedContext.Error = new Exception(this.Error.Message, this.Error.InnerException);
```

Comment:
Why do you need to create a new exception object? Do you care about the exception type? Usually, exceptions are immutable objects. And you do not clone the `InnerException`, it will be shared between the two outer exceptions anyways.